inc/namespace.php

```diff
@@ -125,88 +132,142 @@ function remove_yoast_submenu_page() {
+	$config = Altis\get_config()['modules']['seo']['metadata'] ?? [];
+
+	// Get Opengraph and Twitter card settings. These default to true if the config has been set but no value given to these options.
+	$options['opengraph'] = $config['opengraph'] ?? true;
+	$options['twitter'] = $config['twitter'] ?? true;
+
+	$options['pinterestverify'] = $config['pinterest-verify'] ?? '';
+	$options['facebook_site'] = $config['social-urls']['facebook'] ?? '';
+	$options['twitter_site'] = $config['social-urls']['twitter'] ?? '';
+	$options['instagram_url'] = $config['social-urls']['instagram'] ?? '';
+	$options['linkedin_url'] = $config['social-urls']['linkedin'] ?? '';
+	$options['google_url'] = $config['social-urls']['google'] ?? '';
+	$options['myspace_url'] = $config['social-urls']['myspace'] ?? '';
+	$options['pinterest_url'] = $config['social-urls']['pinterest'] ?? '';
+	$options['youtube_url'] = $config['social-urls']['youtube'] ?? '';
+	$options['wikipedia_url'] = $config['social-urls']['wikipedia'] ?? '';
+
+	// Set fallback image based on config.
+	$options['og_default_image'] = $config['fallback-image'] ?? '';
+	$options['og_default_image_id'] = $config['fallback-image-id'] ?? '';
+
+	// If the fallback image ID was not configured, but we do have a fallback image, get the ID from the URL.
+	if ( empty( $options['og_default_image_id'] ) && $options['og_default_image'] ) {
+		$options['og_default_image_id'] = get_image_id_from_url( $config['fallback-image'] );
+	}
+
+	// These options are only used as fallbacks from the default Home and Front Page SEO options, and possibly not even then.
+	$options['og_frontpage_title'] = $config['opengraph-fallback']['og-frontpage-title'] ?? '';
+	$options['og_frontpage_desc'] = $config['opengraph-fallback']['og-frontpage-desc'] ?? '';
+	$options['og_frontpage_image'] = $config['opengraph-fallback']['og-frontpage-image'] ?? '';
+	$options['og_frontpage_image_id'] = $config['opengraph-fallback']['og-frontpage-image-id'] ?? '';
+
+	// If the frontpage image ID was not configured but we do have a frontpage image, get the ID from the url.
+	if ( empty( $options['og_frontpage_image_id'] ) && $options['og_frontpage_image'] ) {
+		$options['og_frontpage_image_id'] = get_image_id_from_url( $config['opengraph-fallback']['og-frontpage-image'] );
+	}
+
+	return $options;
 }
 
 /**
- * Update twitter metadata to use Tachyon img URL.
+ * Check if we should override the metadata options.
  *
- * @param array $meta Twitter metadata.
+ * Compares the SEO metadata options saved in the config file with the default values. If anything has been changed, returns true.
  *
- * @return array Twitter metadata with image using Tachyon URL, if any.
+ * @return bool True if metadata values have been saved.
  */
-function metadata_img_as_tachyon_twitter( array $meta ) : array {
-	return metadata_img_as_tachyon( $meta, [
-		'resize' => '1200,600', // crop.
-	] );
+function should_override_metadata_options() : bool {
```

Comment:
So as soon as one option is set in config we switch off the admin page? Pity we can't do it more granularly :/

Reply:
Yeah, all of these settings are stored as a single serialized option value, so as soon as you use `pre_option_wpseo_social`, you're overriding anything that might've been saved in the database. In fact, it actually clears out the database options if you switch, which is why I thought removing the admin page was better. You can't make changes that save if you're using the config at all and when you switch, everything that was saved in the config is not stored in the database so it's all empty.